plasma_framework/contracts/src/exits/utils/ExitId.sol

```diff
@@ -35,35 +36,24 @@ library ExitId {
     )
         internal
         pure
-        returns (uint160)
+        returns (uint168)
     {
+        bytes32 outputId;
         if (_isDeposit) {
-            bytes32 hashData = keccak256(abi.encodePacked(_txBytes, _utxoPos.encode()));
-            return _computeStandardExitId(hashData, _utxoPos.outputIndex);
+            outputId = OutputId.computeDepositOutputId(_txBytes, _utxoPos.outputIndex, _utxoPos.encode());
+        } else {
+            outputId = OutputId.computeNormalOutputId(_txBytes, _utxoPos.outputIndex);
         }
 
-        return _computeStandardExitId(keccak256(_txBytes), _utxoPos.outputIndex);
+        return uint168((uint256(outputId) >> (256 - FIRST_BIT)));
```

Comment:
hmmm....shouldn't we first cast to size 168 then shift?

Reply:
No, I think this is correct.
We cast to uint256 to convert from bytes32 to an integer. You can't cast a bytes256 directly to uint168.
Then we shift it to get the _first_ 167 bits of the hash

Do you mean that we could use the _last_ 167 bytes of the hash instead? e.g.
```
return uint168(uint256(outputId)) >> 1;
```

Reply:
ah.....okay now I see that is indeed a correct implementation to get 167 bits out.

and yes, `uint168(uint256(outputId)) >> 1` was indeed what in my mind of getting 167 bits out from the outputId that's why I thought it is a wrong implementation.